app/Http/Controllers/Frontend/SimBriefController.php

```diff
@@ -87,6 +88,10 @@ public function generate(Request $request): RedirectResponse|View
             $user_subfleets = $this->userSvc->getAllowableSubfleets($user)->pluck('id')->toArray();
             $flight_subfleets = $flight->subfleets->pluck('id')->toArray();
 
+            if ((blank($flight_subfleets) || count($flight_subfleets) === 0) && setting('flights.only_company_aircraft', false)) {
+                $flight_subfleets = Subfleet::where(['airline_id' => $flight->airline_id])->pluck('id')->toArray();
+            }
+
```

Comment:
If I'm understanding this correctly, it might be better to change this to...

```
if (setting('flights.only_company_aircraft', false)) {
  // ... run flight_subfleets
} else {
 /// do the default pluck?
}
```

Since in this case, the subfleets are (theoretically) going to be smaller? It's one less query if true

Reply:
There we need to be sure that `$flight->subfleets` is blank to run the query, which allows admins to be able to define subfleets for flights and if it is not done, load airline subfleets to intersect.

And this is done by using the eager loaded relationship collection to create flight subfleet id's array, no additional database/model query being run.

In your suggestion regardless of admin actions for that flight, all airline subfleets will be loaded. Which is not we want actually :)

Finally with the intersection of two arrays (user and flight subfleets) we are creating the "usable" aircraft list down in the code.

So I see no benefit in changing this and using if / else statement there but maybe something like below can be done.

```php
if ($flight->subfleets->count() > 0) {
  $flight_subfleets = $flight->subfleets->pluck('id')->toArray();
} elseif (setting('flights.only_company_aircraft', false)) {
  $flight_subfleets = Subfleet::where(['airline_id' => $flight->airline_id])->pluck('id')->toArray();
} else {
  $flight_subfleets = null;
}
```

If you want I can switch to this and test but I really do not see much performance benefit here as the relationship is already eager loaded with the main query (check line 60).